dissect/target/plugins/os/windows/network.py

```diff
@@ -0,0 +1,383 @@
+import datetime
+
+from dissect.target.exceptions import (
+    RegistryKeyNotFoundError,
+    RegistryValueNotFoundError,
+)
+from dissect.target.helpers.record import WindowsInterfaceRecord
+from dissect.target.plugin import internal
+from dissect.target.plugins.general.network import NetworkPlugin
+
+WINDOWS_EPOCH = datetime.datetime(1601, 1, 1)
+UNIX_EPOCH = datetime.datetime(1970, 1, 1)
+EPOCH_DIFF = (UNIX_EPOCH - WINDOWS_EPOCH).total_seconds()  # Difference in seconds
+
+
+class WindowsNetworkPlugin(NetworkPlugin):
+    @internal
+    def _interfaces(self) -> list[WindowsInterfaceRecord]:
+        for device_info in self._extract_network_devices():
```

Comment:
I'd inline the `self._extract_network_devices` method inside this one. Don't think there's a reason to have it separate.